src/test/java/ru/training/at/hw7/testdata/TestDataLoader.java

```diff
@@ -0,0 +1,79 @@
+package ru.training.at.hw7.testdata;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.testng.annotations.DataProvider;
+
+import java.io.*;
+import java.lang.reflect.Type;
+import java.util.*;
+
+public class TestDataLoader {
+    //Paths to .json and .properties files
+    private String jsonFilePath = "src/test/resources/JDI_ex8_metalsColorsDataSet.json";
+    private static final String credentialsFilePath = "src/test/resources/credentials.properties";
+
+    //All data sets with test values from the json file
+    private Map<String, TestDataSet> allTestData = extractDataFromJson(jsonFilePath);
+
+    //Deserialization of the json into a map
+    private Map<String, TestDataSet> extractDataFromJson(String filePath){
+        Map <String, TestDataSet> extractedData = new HashMap<>();
+        try {
+            Reader reader = new FileReader(jsonFilePath);
+            Gson gson = new Gson();
+            Type type = new TypeToken<HashMap<String, TestDataSet>>(){}.getType();
+            extractedData = gson.fromJson(reader, type);
+        }
+        catch (FileNotFoundException e){
+            System.out.println("Failed to load file from path " + filePath);
+            e.printStackTrace();
+        }
+        return extractedData;
+    }
+
+    //DataProvider method
+    @DataProvider(name = "metalsAndColorsPageTestData")
+    public static Object[][] dpMethod() {
```

Comment:
method name should be more descriptive

Reply:
I agree. This method has been renamed to jsonTestDataProvider(). New name suggests where test values come from